src/core/public/overlays/modal/modal_service.tsx

```diff
@@ -128,6 +149,65 @@ export class ModalService {
 
         return modal;
       },
+      openConfirm: (message: MountPoint | string, options?: OverlayModalConfirmOptions) => {
+        // If there is an active modal, close it before opening a new one.
+        if (this.activeModal) {
+          this.activeModal.close();
+          this.cleanupDom();
+        }
+
+        return new Promise((resolve, reject) => {
+          let resolved = false;
+          const closeModal = (confirmed: boolean) => {
+            resolved = true;
+            modal.close();
+            resolve(confirmed);
+          };
+
+          const modal = new ModalRef();
+          modal.onClose.then(() => {
+            if (this.activeModal === modal) {
+              this.cleanupDom();
+            }
+            // modal.close can be called when opening a new modal/confirm, so we need to resolve the promise in that case.
+            if (!resolved) {
+              closeModal(false);
+            }
+          });
+          this.activeModal = modal;
+
+          const props = {
+            ...options,
+            children:
+              typeof message === 'string' ? (
+                message
+              ) : (
+                <MountWrapper mount={message} className="kbnOverlayMountWrapper" />
+              ),
+            onCancel: () => closeModal(false),
+            onConfirm: () => closeModal(true),
+            cancelButtonText:
+              options?.cancelButtonText ||
+              t.translate('core.overlays.modal.cancel', {
```

Comment:
it's not just `modal`, but `confirmation modal`. how about `core.overlays.confirm.okButton` / `core.overlays.confirm.cancelButton`

Reply:
probably better yes